src/main/java/game/Game.java

```diff
@@ -37,9 +39,27 @@ public void movePlayer(Point2D move) {
         Point2D currentPos = player.getPos();
         Point2D newPos = Point2D.add(currentPos, move);
         boolean traversable = board.isTraversable(newPos);
+        ArrayList<Element> pushables = new ArrayList<>();
+        Point2D movement = new Point2D(newPos.getX() - currentPos.getX(), newPos.getY() - currentPos.getY());
+
+        for (Element object : this.board.getObjectManager().getBoardObjects()) {
+            if (object instanceof PushableElement) {
+                pushables.add(object);
+            }
+        }
 
         if (traversable)  {
             player.setPos(newPos);
+            for (Element pushable : pushables) {
+                if (pushable.getPos().getX() == player.getPos().getX() &
+                        pushable.getPos().getY() == player.getPos().getY()) {
```

Comment:
pushable.getPos() and player.getPos() can be compared using Point2D.equals() method.

Reply:
Oh yeah. Thanks for your suggestion! I will make a change.

Reply:
> 

I think we should stick with comparing the value of the X and Y coordinates. .equals() does not comparing the value of X,Y coordinates.

Reply:
? It does. 

'''
public static boolean equals(Point2D p1, Point2D p2) {
        return (p1.getX() == p2.getX()) & (p1.getY() == p2.getY());
    }
'''